### ISSUE TITLE
Multi language support

### ISSUE THREAD

**Issue by saekort** [saekort/PFS-Scenariotracker#25]
We gotta think on how to implement multi language support.

I think it would need to be a toggle for users somewhere.

For now english is the default and our first extra language oughta be french (due to Flavien) :)

I am thinking we use angular gettext: https://angular-gettext.rocketeer.be/
Then we can generate *.mo and *.po files for the translation and pass them around to people who would like to translate the tracker in different languages.

**Comment by FlavienKnuchel** [comment]
Getting into it right now, gettext is also my preffered solution, so let's get things going!
The flexibility of PO and MO files will make the translation work an easier task!

**Comment by FlavienKnuchel** [comment]
First commit done, branch published, setup OK, now let's translate this!

I am using https://www.transifex.com/ with a free open source account for the translations, so that I can add someone with his email adress, and he can translate directly in browser. Never used that before, testing it right now, I add you (via email) to the traductors, so we can test the feature, and if it is user-frienfly for classic users wanting to help translate

**Comment by saekort** [comment]
This is very interesting. Transifex is effectively a online tool that can get people to generate mo and po for us? I am used to po-edit, but this seems quite user friendly.

**Comment by FlavienKnuchel** [comment]
Yes it is, normally it's a paid subscription, but they are offering their service for free if you can prove you are an open source developper. It's a way to centralise the management of the language translations, to assign people to a certain language, and when they are done, we just recover the po file, process it in gettext, and we're done!

**Comment by saekort** [comment]
So lets see, where do we want to implement a language toggle. I discussed it a bit with @NieskeL and we think there should be two options:

1: While logged in
2: While not logged in

Option 1 needs to look at the logged in user's profile to determine what language has been selected.
Option 2 needs to look at something else (localstorage? I dont like cookies all that much)

We can have a small dropdown with a few flags and perhaps names behind it at the top right. We could also throw that to your profile when logged in, etc.?

**Comment by FlavienKnuchel** [comment]
The two options are viable, I think localstorage would do the job perfectly. 

Interface speaking:
logged out:  a small dropdown on the top bar (next to other) would do the job perfectly, and I don't think flags are a necessity (as a language is no linked only to one country. like in switzerland we speak german french and italian),  but we could choose the main flag for each language and it maybe would be helping people to locate their language faster. I can't really make up my mind on this, I think I would have to see it to choose. 

logged in: We surely can move this function inside the profile page, showing the exact same dropdown on top of it. Or maybe save it inside the user's profile, but it feels a bit overkill.

in the end, I feel like adding just a dropdown menu on the top, save in localstorage, keep it simple.

**Comment by saekort** [comment]
Hmm yeah good point about the profile. That is if we never want to save the language for something where we would need their language. For now lets assume we don't need it and go for localstorage.

As for the flags I would say still go for it, my reasoning is the following: There is not a lot of space in the menu and we want to keep it as small as possible (before someone clicks on a dropdown). If we show ONLY the flag it is much smaller then the language name. If we then, once someone clicks on it, show the flag + the name it would be the best of both worlds no?

**Comment by FlavienKnuchel** [comment]
Yes, I don't know if language is something we want to store in the user. In my sight that's not necessary, as all the content will be translated, and they can change language on the fly :)

Is that really an improvement to add an interaction to the user, to gain width on a drop-down? I'd say we put falg + language name, and call it a day :D And for the active language, we could just display the two-letters initials ?

**Comment by saekort** [comment]
Flag + 2 letter initials for the active one, flag + full name for the dropdown sounds good.
